x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/inference/EnsembleInferenceModel.java

```diff
@@ -142,9 +132,7 @@ private InferenceResults innerInfer(double[] features, InferenceConfig config, M
         if (preparedForInference == false) {
             throw ExceptionsHelper.serverError("model is not prepared for inference");
         }
-        LOGGER.debug("Inference called with feature names [{}]",
-            featureNames == null ? "<null>" : Strings.arrayToCommaDelimitedString(featureNames));
-        assert featureNames != null && featureNames.length > 0;
+        LOGGER.debug("Inference called with feature names [{}]", Strings.arrayToCommaDelimitedString(featureNames));
```

Comment:
Should we make this debug message take a runnable? I think it's called for each doc so it might be worth avoiding running all this if logging is not debug.

Reply:
DEFINITELY! It should be a runnable.